### ISSUE TITLE
Spring Boot Reactive and Cloud won't start due to missing ReactiveWebServerFactory bean

### ISSUE THREAD

**Issue by aamzerin** [spring-cloud/spring-cloud-netflix#2824]
Hello,

I'm facing some problems when using Spring Boot Reactive with spring cloud.

It seems that the application (microsrevice) can't start due to this problem:

`org.springframework.context.ApplicationContextException: Unable to start reactive web server; nested exception is org.springframework.context.ApplicationContextException: Unable to start ReactiveWebApplicationContext due to missing ReactiveWebServerFactory bean. at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.onRefresh(ReactiveWebServerApplicationContext.java:76) ~[spring-boot-2.0.0.RELEASE.jar:2.0.0.RELEASE] at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:543) ~[spring-context-5.0.4.RELEASE.jar:5.0.4.RELEASE] at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:61) ~[spring-boot-2.0.0.RELEASE.jar:2.0.0.RELEASE] at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752) [spring-boot-2.0.0.RELEASE.jar:2.0.0.RELEASE] at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:388) [spring-boot-2.0.0.RELEASE.jar:2.0.0.RELEASE] at org.springframework.boot.SpringApplication.run(SpringApplication.java:327) [spring-boot-2.0.0.RELEASE.jar:2.0.0.RELEASE] at org.springframework.boot.SpringApplication.run(SpringApplication.java:1246) [spring-boot-2.0.0.RELEASE.jar:2.0.0.RELEASE] at org.springframework.boot.SpringApplication.run(SpringApplication.java:1234) [spring-boot-2.0.0.RELEASE.jar:2.0.0.RELEASE] at io.aamzerin.micro.start.server.ServerApp.main(ServerApp.java:22) [classes/:na] Caused by: org.springframework.context.ApplicationContextException: Unable to start ReactiveWebApplicationContext due to missing ReactiveWebServerFactory bean. at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.getWebServerFactory(ReactiveWebServerApplicationContext.java:125) ~[spring-boot-2.0.0.RELEASE.jar:2.0.0.RELEASE] at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.createWebServer(ReactiveWebServerApplicationContext.java:99) ~[spring-boot-2.0.0.RELEASE.jar:2.0.0.RELEASE] at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.onRefresh(ReactiveWebServerApplicationContext.java:73) ~[spring-boot-2.0.0.RELEASE.jar:2.0.0.RELEASE] ... 8 common frames omitted`

I'm using Spring cloud Finchley.M9 and Spring Boot 2.

- Project repo: https://github.com/aamzerin/client-server-micro-starter

- Infrastructure repo: https://github.com/aamzerin/micro-arch-infrastructure

Please feel free to ask for more information.

Thank you!

**Comment by ryanjbaxter** [comment]
Can you simplify this down to 1 or 2 apps to reproduce the problem?

**Comment by aamzerin** [comment]
Hello,

   I just pushed a simple application that reproduces the problem to this repo:
   [https://github.com/aamzerin/reactive-cloud](url)

   I also want to point out that I was able to resolve the problem by upgrading Spring Boot parent from `2.0.0.RELEASE` to `2.0.1.RELEASE `. 
  
  So it seems that Spring Boot 2.0.0 isn't compatible with Spring Cloud Finchley.M9.

  I didn't push the fix to keep the error reproducible.

Thanks for your time!
